### PR TITLE
Externalized strings

### PR DESCRIPTION
**Pull request by evannwu20**
Signed-off-by: Evann Wu <Evann.Wu@ibm.com>

### CODE REVIEW

File: resources/testProfileData.example.ts [hunk 1]
```diff
@@ -2,10 +2,10 @@ import { IProfile } from "@brightside/imperative";
...
 export const profile: IProfile = {
     type : "zosmf",
-    host: "",
-    port: 0,
-    user: "",
-    password: "",
+    host: "http://tvt4119.svl.ibm.com",
+    port: 443,
+    user: "zoweew",
+    password: "waziusr",
```
Comment (kristinochka) [inline resources/testProfileData.example.ts]: This file should not be changed, it's an example. If you want to run integration tests you need to create testProfileData.ts. Here's README with more info https://github.com/phaumer/vscode-extension-for-zowe/tree/master/docs.
Also, don't put usernames and password in source code.

**Review by kristinochka — CHANGES_REQUESTED** [review]

**Comment by kristinochka** [comment]
Should there be a strings file/folder for English? i18n folder

File: resources/testProfileData.example.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { IProfile } from "@brightside/imperative";
 
 export const profile: IProfile = {
-    type : "zosmf",
```
Comment (kristinochka) [inline resources/testProfileData.example.ts]: This value shouldn't be deleted.

File: src/ZoweNode.ts [hunk 3]
```diff
@@ -67,8 +71,8 @@ export class ZoweNode extends vscode.TreeItem {
         }
 
         if (!this.mLabel) {
-            vscode.window.showErrorMessage("Invalid node");
-            throw Error("Invalid node");
+            vscode.window.showErrorMessage(localize("ZoweNode.ZoweNode.getChildren.error.message.invalidNode.text", "Invalid node"));
```
Comment (kristinochka) [inline src/ZoweNode.ts]: Could you find a way to make these keys shorter?

**Review by kristinochka — CHANGES_REQUESTED** [review]

**Comment by kristinochka** [comment]
Do i18n/.DS_Store files need to be checked in?

**Comment by kristinochka** [comment]
Unit tests are failing.

File: i18n/sample/package.i18n.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
+{
+<<<<<<< HEAD
```
Comment (kristinochka) [inline i18n/sample/package.i18n.json]: You forgot to remove the merge conflict markers in this file.

**Comment by kristinochka** [comment]
So couple of things need to be done here:

- [x] Ensure package.json file is formatted correctly so only the strings show as changed.
- [x]  Shorten the strings even more, for example:
`localize("addSession.message.error.loadProfiles",` vs 
`localize("addSession.error.loadProfiles",`
and 
`localize("addSession.message.noProfilesAdd", "No more profiles to add"))`vs
`localize("addSession.noProfilesAdd", "No more profiles to add"))`
- [x] `docs/README.md` needs to be updated to let developers know how to handle new strings. Add new section for localization
- [x] Resolve merge conflicts and remove merge conflict markers.

File: src/extension.ts [hunk 5]
```diff
@@ -713,16 +723,23 @@ export async function deleteDataset(node: ZoweNode, datasetProvider: DatasetTree
...
-            vscode.window.showInformationMessage(`Unable to find file: ${label} was probably already deleted.`);
+        log.error(localize("deleteDataSet.delete.log.error", "Error encountered when deleting data set! ") + JSON.stringify(err));
+        if (err.message.includes(localize("deleteDataSet.error.notFound", "not found"))) {
+<<<<<<< HEAD
+            vscode.window.showInformationMessage(localize("deleteDataSet.notFound.error.message1", "Unable to find file: ") + label +
+            localize("deleteDataSet.notFound.error.message2", " was probably already deleted."));
+=======
+            vscode.window.showInformationMessage(localize("deleteDataSet.notFound.error.message1", "Unable to find file:") + label +
+            localize("deleteDataSet.notFound.error.message2", "was probably already deleted."));
+>>>>>>> e76d5ef5182b4b1eb8d6d3202df90831d609d439
```
Comment (kristinochka) [inline src/extension.ts]: You forgot to remove merge conflict markers here as well.

**Review by kristinochka — CHANGES_REQUESTED** [review]

File: docs/README.md [hunk 6]
```diff
@@ -108,4 +108,34 @@ There is no required structure for the mainframe data sets under `orPattern`.
...
+
+#### Removing Strings
+- When removing strings, remove the key/value pair from its corresponding .json file (either package.nls.json or _filename_.i18n.json)
+
+#### Changing Strings
+- When changing strings, make the same change to the key/value pair from its corresponding .json file (either package.nls.json or _filename_.i18n.json)
+
+### How to Donate Translations
+1. Click [here](https://translate.zanata.org/?dswid=8786) and follow instructions under Sign Up heading to sign up to Zatana.
+2. Send an email to evann.wu@ibm.com with the email heading as ZATANA TRANSLATOR REQUEST and include the following information in the body of the email.
```
Comment (kristinochka) [inline docs/README.md]: We need to think about who will be an admin of this account when you leave.

Comment (evannwu20) [reply docs/README.md]: Maybe Lauren? She's worked on localization previously.

File: src/extension.ts [hunk 7]
```diff
@@ -60,7 +63,7 @@ export async function activate(context: vscode.ExtensionContext) {
         Logger.initLogger(loggerConfig);
 
         log = Logger.getAppLogger();
-        log.debug("Initialized logger from VSCode extension");
+        log.debug(localize("log.debug", "Initialized logger from VSCode extension"));
```
Comment (kristinochka) [inline src/extension.ts]: This should be re-named  to something more specific even though it's at top level outside of a function, for example: "initialization.log.debug"

File: src/extension.ts [hunk 8]
```diff
@@ -69,7 +72,7 @@ export async function activate(context: vscode.ExtensionContext) {
         ussFileProvider = new USSTree();
         await ussFileProvider.addSession();
     } catch (err) {
-        log.error("Error encountered while activating and initializing logger! " + JSON.stringify(err));
+        log.error(localize("log.error", "Error encountered while activating and initializing logger! ") + JSON.stringify(err));
```
Comment (kristinochka) [inline src/extension.ts]: Same as above, "initialization.log.error"

**Review by kristinochka — CHANGES_REQUESTED** [review]

**Comment by kristinochka** [comment]
I think everything looks pretty good! Only 2 small things above

**Pull request closed without merging**